R/EnrichDepletTF.R

```diff
@@ -0,0 +1,417 @@
+  upCorGenes <- sysCorGenes(exProfS = exProfS,    #the genes that were
+                            nUpMost = nUpMost,    #Upmost correlated.
+                            direction = CorDirection,
+                            multipleTests = multipleTests,
+                            alpha = alpha)
+
+  if (sum( ! is.na(upCorGenes)) >= 3) {
+    #calculating enrichment and depletion:
+    corEnrichDep <- as.data.frame(matrix(data = NA, nrow = length(TfSym), ncol = 8),
+                                   stringsAsFactors = F)
+    colnames(corEnrichDep) <- c("TF", "Enrichment", "Enrichment_P_value",
+                                "Depletion", "Depletion_P_value",
+                                "BH_enrichment","BH_depletion",
+                                "Bonferroni")
+    corEnrichDep$TF <- TfSym
+
+    for (i in seq_along(TfSym)){
+      #Calculating enrichment and p value
+      GenesBindTf <- GTRDtf[[TfSym[i]]]
+      CorGenesBindTf <- upCorGenes[which(upCorGenes %in% GenesBindTf)]
+      a <- length(CorGenesBindTf) #Correlates and binds TF
+      b <- length(GenesBindTf) - a #not correlates and binds TF
+      c <- length(upCorGenes) - a #Correlates and don't bind
+      d <- (length(GeneSym) - length(upCorGenes)) - b #Not correlates and not binds
+
+      tmpFisher <- stats::fisher.test(matrix(c(a, b, c, d), nrow = 2),
+                                      alternative = "greater")
+      corEnrichDep$Enrichment[i] <- tmpFisher$estimate
+      corEnrichDep$Enrichment_P_value[i] <- tmpFisher$p.value
+
+      #Calculating depletion. and p value
+      tmpFisher <- stats::fisher.test(matrix(c(c, d, a, b), nrow = 2),
+                                      alternative = "greater")
+      corEnrichDep$Depletion[i] <- tmpFisher$estimate
+      corEnrichDep$Depletion_P_value[i] <- tmpFisher$p.value
+    }
+
+    #Calculating the cutoffs for each multi-test correction approach
+    nExperiments <- length(TfSym)
+    corEnrichDep <- corEnrichDep[order(corEnrichDep$Enrichment_P_value,
+                                         decreasing = FALSE),] #order according to
+                                                               #increasing p values
+    corEnrichDep$BH_enrichment <- (order(corEnrichDep$Enrichment_P_value,
+                                          decreasing = FALSE) / nExperiments) * alpha
+    corEnrichDep <- corEnrichDep[order(corEnrichDep$Depletion_P_value,
+                                         decreasing = FALSE),] #order according to
+                                                               #increasing P values
+    corEnrichDep$BH_depletion <- (order(corEnrichDep$Depletion_P_value,
+                                         decreasing = FALSE) / nExperiments) * alpha
+    corEnrichDep$Bonferroni <- alpha / nExperiments #Bonferroni cutoff value.
+
+    return(corEnrichDep)
+
+  } else {
+    return (print("Error: less then 3 significantly correlated genes"))
```

Comment:
As above. Also: inability to calculate enrichment because of lack of genes is not an "error", but expected. Perhaps return `NULL` (or even `NaN` - because it's a mathematically undefined situation) ... and use `message()` instead.